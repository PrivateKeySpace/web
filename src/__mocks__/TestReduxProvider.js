import React from 'react'
import PropTypes from 'prop-types'
import { createStore } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'
import rootReducer from '../store/rootReducer'

const displayName = '__mocks__/TestReduxProvider'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ state: PropTypes.object, children: PropTypes.any }),
  defaultProps({ state: {} })
)

function TestReduxProvider (props) {
  const { state, children } = props
  const store = createStore(rootReducer, state)

  return (
    <ReduxProvider store={store}>
      {children}
    </ReduxProvider>
  )
}

export default enhance(TestReduxProvider)
