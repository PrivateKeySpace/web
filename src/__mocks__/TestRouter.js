import React from 'react'
import PropTypes from 'prop-types'
import { MemoryRouter } from 'react-router'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const displayName = '__mocks__/TestRouter'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ path: PropTypes.string, children: PropTypes.any }),
  defaultProps({ path: '/' })
)

function TestRouter (props) {
  const { path, children } = props

  return (
    <MemoryRouter initialEntries={[path]} initialIndex={0}>
      {children}
    </MemoryRouter>
  )
}

export default enhance(TestRouter)
