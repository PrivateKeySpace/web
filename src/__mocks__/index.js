export { default as TestRouter } from './TestRouter'
export { default as TestReduxProvider } from './TestReduxProvider'
