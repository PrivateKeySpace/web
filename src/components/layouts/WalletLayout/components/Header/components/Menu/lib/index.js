export { default as mapDispatch } from './mapDispatch'
export { default as handlers } from './handlers'
export { default as stateSpec } from './stateSpec'
