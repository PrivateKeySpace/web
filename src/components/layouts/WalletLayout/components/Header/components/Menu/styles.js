import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../../../theme/constants'

const styles = StyleSheet.create({
  dropdownButton: {
    position: 'relative',
    width: 32,
    userSelect: 'none'
  },
  icon: {
    color: colors.white,
    opacity: 0.5,
    ':hover': {
      opacity: 1
    }
  },
  iconActive: {
    opacity: 1
  },
  dropdownButtonActive: {
    backgroundColor: '#131a1e'
  },
  dropdownMenu: {
    cursor: 'default',
    backgroundColor: '#131a1e',
    width: 165,
    position: 'absolute',
    top: '100%',
    right: 0
  }
})

export default styles
