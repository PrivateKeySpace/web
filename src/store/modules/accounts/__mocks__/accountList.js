const accountList = [
  {
    'path': [
      2147483697,
      2147483648,
      2147483648
    ],
    'address': '3Pc8vh9xUopMmiBW1yDAm3h1aqqbmmv11p',
    'addressPath': [
      2147483697,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6Ca3FEzN1SemuKNujpAeyDLXk7DXXtgpUs5HHLrG5brjVQm7ib3SLqMKfWWQZset7HMm7wM58wVGEJm5eG6rkxEcYvtvExYaM9T7jxkmmMp',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': true
  },
  {
    'path': [
      2147483692,
      2147483648,
      2147483648
    ],
    'address': '1Bswvjfoa1p1kMkwE4C8KAPbYLSKfg1Nni',
    'addressPath': [
      2147483692,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6D4zdJZuWGKQK5qMyj7reDcapRE6StaVTGS3xWCAjRyLfne84tGxUK1yYNrUWFY8gYtJA7xd3s97o1uiPkFEhpdsuDsNkLnj5iHyAbCEnRv',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': false
  }
]

export default accountList
