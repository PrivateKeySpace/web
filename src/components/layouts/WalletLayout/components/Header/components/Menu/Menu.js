import React from 'react'
import onClickOutside from 'react-onclickoutside'
import { connect } from 'react-redux'
import { compose, setDisplayName, pure, withHandlers, withState } from 'recompose'
import { css } from 'aphrodite/no-important'
import classNames from 'classnames'
import sx from './styles'
import tx from '../../../../../../../theme/styles'
import { mapDispatch, handlers, stateSpec } from './lib'
import { MenuItem } from './components'

const displayName = 'components/layouts/WalletLayout/components/Header/components/Menu'

const enhance = compose(
  setDisplayName(displayName),
  connect(undefined, mapDispatch),
  ...stateSpec.map(statePropDefinition => withState(...statePropDefinition)),
  withHandlers(handlers),
  onClickOutside,
  pure
)

function Menu (props) {
  const { isMenuOpen, toggleMenu, handleSignOut } = props
  return (
    <div onClick={toggleMenu} className={css(tx.flex, tx.alignItemsCenter, tx.justifyContentCenter, tx.px2, tx.ml1, tx.height100, tx.cursorPointer, sx.dropdownButton, isMenuOpen && sx.dropdownButtonActive)}>
      <span className={classNames('lnr lnr-chevron-down', css(sx.icon, isMenuOpen && sx.active))} />
      {
        isMenuOpen &&
        <div className={css(tx.flex, tx.flexDirectionColumn, tx.py2, sx.dropdownMenu)}>
          <MenuItem icon='user'>My Profile</MenuItem>
          <MenuItem icon='exit' onClick={handleSignOut}>Log Out</MenuItem>
        </div>
      }
    </div>
  )
}

export default enhance(Menu)
