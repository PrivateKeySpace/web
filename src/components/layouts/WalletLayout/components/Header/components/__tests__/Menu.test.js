import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { TestReduxProvider } from '../../../../../../../__mocks__'
import Menu from '../Menu'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const footer = render(
    <TestReduxProvider>
      <Menu />
    </TestReduxProvider>
  )
  expect(footer).toMatchSnapshot()
})
