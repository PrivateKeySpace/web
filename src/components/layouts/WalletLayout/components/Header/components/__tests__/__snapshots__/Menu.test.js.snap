// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-px2_6ebohg-o_O-ml1_1dn7v29-o_O-height100_1yfus1e-o_O-cursorPointer_13xlah4-o_O-dropdownButton_1w6iw80"
>
  <span
    class="lnr lnr-chevron-down icon_ko4whp"
  >
    
  </span>
</div>
`;
